packages/client/src/actions/applyAction.ts

```diff
@@ -93,31 +100,35 @@ export async function applyAction<
   }
 }
 
-function remapActionParams<AD extends ActionDefinition<any, any>>(
+async function remapActionParams<AD extends ActionDefinition<any, any>>(
   params: OsdkActionParameters<AD["parameters"]> | undefined,
-): Record<string, DataValue> {
+  client: MinimalClient,
+): Promise<Record<string, DataValue>> {
   if (params == null) {
     return {};
   }
 
   const parameterMap: { [parameterName: string]: any } = {};
-  const remappedParams = Object.entries(params).reduce((acc, [key, value]) => {
-    acc[key] = toDataValue(value);
-    return acc;
-  }, parameterMap);
+  const remappedParams = Object.entries(params).reduce(
+    async (promisedAcc, [key, value]) => {
+      const acc = await promisedAcc;
+      acc[key] = await toDataValue(value, client);
+      return acc;
+    },
+    Promise.resolve(parameterMap),
```

Comment:
This is a fascinating trick that might be too clever. Compare to straight forward:
```ts
const parameterMap: { [parameterName: string]: unknown } = {};
for (const key of Object.keys(params)) {
  parameterMap[key] = await toDataValue(value, client);
}
```